ls1intum/Artemis
# Enable LTI for modeling, text and file upload exercises

Until now, LTI and the use of exercises in online courses was limited to programming exercises.
This PR enables LTI for online courses for modeling, text and file upload exercises.

When tutors submit an assessment for text, modeling, file upload or programming exercises, the score is immediately reported to edX. This means Artemis ignores the assessment due date in this case, because otherwise we would need to implement a scheduler to report the results to edX, which is not really necessary in my opinion, as it is rather difficult and might be error prone in case of system restarts. We could still implement something like this in the future, but for now this is not needed!

### Checklist

### Steps for Testing
Only for SEECx tutors:

1. Create a programming, text, modeling and file upload exercise on Artemis TS1 (make sure the course is configured as online course). Test programming exercises with automatic and manual assessment
2. Reference the exercises on edX in an unreleased section
3. Navigate from edX to Artemis (so that Artemis has a proper LTI URL for the exercise and the user), start the exercise, submit something
4. Assess the text/modeling/file upload/programming exercise manually on Artemis after the due date
5. Check that the score (set in the assessment) is successfully reported to edX and is shown on the edX page

### Test Coverage
no change
